src/f2py/pyf_preprocessor.py

```diff
@@ -69,12 +69,16 @@
             # We now have to check for real/integer types and process:
             if "kind=inttype" in orig_lines[i]:
                 orig_lines[i] = orig_lines[i].replace("kind=inttype", "kind=4")
+
             if "real(kind=realtype)" in orig_lines[i]:
                 if mode == "real":
                     orig_lines[i] = orig_lines[i].replace("kind=realtype", "kind=8")
                 else:
                     orig_lines[i] = orig_lines[i].replace("real(kind=realtype)", "complex(kind=8)")
-                # end if
+
+            if "real(kind=alwaysrealtype)" in orig_lines[i]:
```

Comment:
is this the fix required to have always real type options passed from python? does this require any other changes to the code?